### PR TITLE
Login: Activity Indicator and handle response

### PR DESCRIPTION
**Pull request by mcolombo87**
## Summary

Login Screen part 3. 

- Activity Indicator while the user waiting for a response.
- Response handler
- HOC and a lot of little fixes or little changes.
- Errors messages for user.

## Screenshots
IMAGE ALT TEXT HERE

## Trello Card
https://trello.com/c/igCW825w/35-patterns-ii-hoc

### CODE REVIEW

File: src/redux/login/actions.js [hunk 1]
```diff
@@ -8,18 +8,25 @@ export const actions = {
...
     const response = await login(email, pwd);
     if (response.ok) {
-      // TODO: all actions make the same thing, this will be change when I need store data (in another card)
       dispatch({
         type: actions.SIGN_IN_SUCCESS,
-        payload: response.ok
+        payload: {
+          token: response.headers['access-token'],
+          waitingResponse: false,
+          error: ''
```
Comment (guidoprinc) [inline src/redux/login/actions.js:20]: i would suggest null

Comment (guidoprinc) [reply src/redux/login/actions.js:20]: so you can check it easier in validations

File: src/redux/login/reducer.js [hunk 2]
```diff
@@ -1,20 +1,26 @@
 const initialState = {
-  token: undefined
+  token: '',
+  waitingResponse: false,
+  error: ''
```
Comment (guidoprinc) [inline src/redux/login/reducer.js:4]: same as above

File: src/redux/login/actions.js (hunk 1 above)
Comment (mcolombo87) [reply src/redux/login/actions.js]: Done!

File: src/redux/login/reducer.js (hunk 2 above)
Comment (mcolombo87) [reply src/redux/login/reducer.js]: Done!

**Pull request merged**